Index: apt-0.8.10.3/buildlib/environment.mak.in
===================================================================
--- apt-0.8.10.3.orig/buildlib/environment.mak.in
+++ apt-0.8.10.3/buildlib/environment.mak.in
@@ -37,8 +37,10 @@ PO4A := @PO4A@
 # Gettext settings
 GMSGFMT = @GMSGFMT@
 XGETTEXT = @XGETTEXT@
-MSGCOMM:=$(dir $(XGETTEXT))/msgcomm
-MSGMERGE:=$(dir $(XGETTEXT))/msgmerge
+#MSGCOMM:=$(dir $(XGETTEXT))/msgcomm
+MSGCOMM:=/usr/bin/msgcomm
+#MSGMERGE:=$(dir $(XGETTEXT))/msgmerge
+MSGMERGE:=/usr/bin/msgmerge
 BASH = @BASH@
 
 # Various library checks
@@ -59,7 +61,7 @@ NEED_SOCKLEN_T_DEFINE = @NEED_SOCKLEN_T_
 
 # Shared library things
 HOST_OS = @host_os@
-ifneq ($(words $(filter gnu% linux-gnu% kfreebsd-gnu% %-gnu,$(HOST_OS))),0)
+ifneq ($(words $(filter solaris% gnu% linux-gnu% kfreebsd-gnu% %-gnu,$(HOST_OS))),0)
    SONAME_MAGIC=-Wl,-soname -Wl,
    LFLAGS_SO=
 else
