Index: apt-0.8.10.3+squeeze1/doc/Doxyfile.in
===================================================================
--- apt-0.8.10.3+squeeze1/doc/Doxyfile.in
+++ apt-0.8.10.3+squeeze1/doc/Doxyfile.in
@@ -38,7 +38,7 @@ PROJECT_NUMBER         = @VERSION@
 # If a relative path is entered, it will be relative to the location
 # where doxygen was started. If left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = ../build/doc/doxygen
+OUTPUT_DIRECTORY       = 
 
 # If the CREATE_SUBDIRS tag is set to YES, then doxygen will create
 # 4096 sub-directories (in 2 levels) under the output directory of each output
