=== modified file 'apt-pkg/aptconfiguration.cc'
--- apt-0.9.5.1/apt-pkg/aptconfiguration.cc	2012-05-20 16:58:37 +0000
+++ apt-0.9.5.1/apt-pkg/aptconfiguration.cc	2012-07-20 05:42:50 +0000
@@ -166,8 +166,8 @@
 				continue;
 			builtin.push_back(c);
 		}
+		closedir(D);
 	}
-	closedir(D);
 
 	// FIXME: Remove support for the old APT::Acquire::Translation
 	// it was undocumented and so it should be not very widthly used

