=== modified file 'configure.in'
--- apt-0.9.5.1/configure.in	2012-05-24 15:27:34 +0000
+++ apt-0.9.5.1/configure.in	2012-07-14 19:57:28 +0000
@@ -20,9 +20,9 @@
 PACKAGE="apt"
 PACKAGE_VERSION="0.9.5.1"
 PACKAGE_MAIL="APT Development Team <deity@lists.debian.org>"
-AC_DEFINE_UNQUOTED(PACKAGE,"$PACKAGE")
+AC_DEFINE_UNQUOTED(PACKAGE,"$PACKAGE","Package name")
 AC_DEFINE_UNQUOTED(PACKAGE_VERSION,"$PACKAGE_VERSION")
-AC_DEFINE_UNQUOTED(PACKAGE_MAIL,"$PACKAGE_MAIL")
+AC_DEFINE_UNQUOTED(PACKAGE_MAIL,"$PACKAGE_MAIL","Package mail")
 AC_SUBST(PACKAGE)
 AC_SUBST(PACKAGE_VERSION)
 AC_SUBST(PACKAGE_MAIL)
@@ -72,7 +72,7 @@
        [#include <db.h>],
        [int r, s, t; db_version(&r, &s, &t);]
       )],
-      [AC_DEFINE(HAVE_BDB)
+      [AC_DEFINE(HAVE_BDB, [1], [Have dh.h])
        BDBLIB="-ldb"
        AC_MSG_RESULT(yes)],
       [BDBLIB=""
@@ -97,14 +97,14 @@
 	AC_MSG_ERROR([failed: Need libz]))
 AC_SUBST(HAVE_ZLIB)
 if test "x$HAVE_ZLIB" = "xyes"; then
-	AC_DEFINE(HAVE_ZLIB)
+	AC_DEFINE(HAVE_ZLIB, [1], [Have zlib.h])
 fi
 
 HAVE_BZ2=no
 AC_CHECK_LIB(bz2, BZ2_bzopen,[AC_CHECK_HEADER(bzlib.h, [HAVE_BZ2=yes], [])], [])
 AC_SUBST(HAVE_BZ2)
 if test "x$HAVE_BZ2" = "xyes"; then
-	AC_DEFINE(HAVE_BZ2)
+	AC_DEFINE(HAVE_BZ2, [1], [Have bzlib.h])
 fi
 
 dnl Converts the ARCH to be something singular for this general CPU family
@@ -117,7 +117,7 @@
    AC_MSG_ERROR([failed: use --host= or output from dpkg-architecture])
 fi
 AC_MSG_RESULT($archset)
-AC_DEFINE_UNQUOTED(COMMON_ARCH,"$archset")
+AC_DEFINE_UNQUOTED(COMMON_ARCH,"$archset","Common Arch")
 
 dnl Single Unix Spec statvfs
 AC_CHECK_FUNC(statvfs,[HAVE_STATVFS=yes])
@@ -125,8 +125,8 @@
 
 dnl Arg, linux and bsd put their statfs function in different places
 if test x"$HAVE_STATVFS" != x"yes"; then
-   AC_EGREP_HEADER(statfs,sys/vfs.h,[AC_DEFINE(HAVE_VFS_H)],[
-   AC_EGREP_HEADER(statfs,sys/mount.h,[AC_DEFINE(HAVE_MOUNT_H)],[AC_MSG_ERROR(failed: Need statvfs)])
+   AC_EGREP_HEADER(statfs,sys/vfs.h,[AC_DEFINE(HAVE_VFS_H, [1], [Have sys/vfs.h])],[
+   AC_EGREP_HEADER(statfs,sys/mount.h,[AC_DEFINE(HAVE_MOUNT_H, [1], [Have sys/mount.h])],[AC_MSG_ERROR(failed: Need statvfs)])
    ])
 fi
 
@@ -135,7 +135,7 @@
      #include <sys/vfs.h>])
 
 dnl We should use the real timegm function if we have it.
-AC_CHECK_FUNC(timegm,AC_DEFINE(HAVE_TIMEGM))
+AC_CHECK_FUNC(timegm,AC_DEFINE(HAVE_TIMEGM, [1], [Have timegm]))
 AC_SUBST(HAVE_TIMEGM)
 
 dnl Check the architecture
@@ -144,7 +144,7 @@
 dnl HP-UX sux..
 AC_MSG_CHECKING(for missing socklen_t)
 AC_EGREP_HEADER(socklen_t, sys/socket.h,[AC_MSG_RESULT(no)],[
-	AC_DEFINE(NEED_SOCKLEN_T_DEFINE)
+	AC_DEFINE(NEED_SOCKLEN_T_DEFINE, [1], [Need socklen_t])
 	NEED_SOCKLEN_T_DEFINE=yes
         AC_MSG_RESULT(missing.)])
 AC_SUBST(NEED_SOCKLEN_T_DEFINE)
@@ -186,7 +186,7 @@
 ALL_LINGUAS="bg bs ca cs cy da de dz el es eu fi fr gl hu it ja ko ku nb nl nn pl pt_BR pt ro ru sk sl sv th tl uk vi zn_CN zh_TW"
 AM_GNU_GETTEXT(external)
 if test x"$USE_NLS" = "xyes"; then
-   AC_DEFINE(USE_NLS)
+   AC_DEFINE(USE_NLS, [1], [Use NLS])
 fi
 AC_SUBST(USE_NLS)
 AC_PATH_PROG(BASH, bash)

