--- ImageMagick-6.3.4/coders/png.c.orig	Mon Aug 23 12:50:12 2010
+++ ImageMagick-6.3.4/coders/png.c	Mon Aug 23 12:51:35 2010
@@ -74,8 +74,12 @@
 #include "magick/transform.h"
 #include "magick/utility.h"
 #if defined(HasPNG)
-#include "png.h"
+#include "libpng14/png.h"
 #include "zlib.h"
+#if PNG_LIBPNG_VER >= 10400
+#define trans_values trans_color
+#define trans trans_alpha
+#endif
 
 /* ImageMagick differences */
 #define first_scene scene
--- ImageMagick-6.3.4/configure.ac.orig	2012-06-02 13:26:48.438064148 -0700
+++ ImageMagick-6.3.4/configure.ac	2012-06-02 13:28:40.631971988 -0700
@@ -1487,7 +1487,7 @@
     AC_MSG_RESULT(-------------------------------------------------------------)
     failed=0;
     passed=0;
-    AC_CHECK_HEADER(png.h,passed=`expr $passed + 1`,failed=`expr $failed + 1`,)
+    AC_CHECK_HEADER(libpng14/png.h,passed=`expr $passed + 1`,failed=`expr $failed + 1`,)
     AC_CHECK_LIB(png,png_get_io_ptr,passed=`expr $passed + 1`,failed=`expr $failed + 1`,)
     AC_MSG_CHECKING(if PNG package is complete)
     if test $passed -gt 0; then
@@ -1495,7 +1495,7 @@
             AC_MSG_RESULT(no -- some components failed test)
             have_png='no (failed tests)'
         else
-            PNG_LIBS='-lpng'
+            PNG_LIBS='-lpng14'
             LIBS="$PNG_LIBS $LIBS"
             AC_DEFINE(HasPNG,1,Define if you have PNG library)
             AC_MSG_RESULT(yes)
--- ImageMagick-6.3.4/Makefile.am.bak	2012-06-02 13:50:25.302842904 -0700
+++ ImageMagick-6.3.4/Makefile.am	2012-06-02 13:50:41.754026227 -0700
@@ -21,7 +21,7 @@
 
 AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)
 
-ACLOCAL_AMFLAGS = -Iltdl/m4 -I m4
+ACLOCAL_AMFLAGS = -I m4
 
 MODULECOMMONFLAGS = -no-undefined -module -avoid-version
 
